Backpack/Icon/Classes/BPKIconView.m

```diff
@@ -59,13 +59,45 @@ - (void)setIconName:(nullable BPKIconName)iconName {
         _iconName = [iconName copy];
 
         if (iconName) {
-            super.image = [BPKIcon templateIconNamed:iconName size:self.size];
+            super.image = [self imageWithIconName:iconName size:self.size flipsForRightToLeft:self.flipsForRightToLeft];
         } else {
             super.image = nil;
         }
     }
 }
 
+- (void)setSize:(BPKIconSize)size {
+    BPKAssertMainThread();
+    if (size != _size) {
+        _size = size;
+        
+        if (self.iconName) {
+            super.image = [self imageWithIconName:self.iconName size:size flipsForRightToLeft:self.flipsForRightToLeft];
+        }
+    }
+}
+
+- (void)setFlipsForRightToLeft:(BOOL)flipsForRightToLeft {
+    BPKAssertMainThread();
+    if (flipsForRightToLeft != _flipsForRightToLeft) {
+        _flipsForRightToLeft = flipsForRightToLeft;
+        
+        if (self.iconName) {
+            super.image = [self imageWithIconName:self.iconName size:self.size flipsForRightToLeft:flipsForRightToLeft];
+        }
+    }
+}
+
+- (UIImage *)imageWithIconName:(BPKIconName)iconName size:(BPKIconSize)size flipsForRightToLeft:(BOOL)flipsForRightToLeft {
+    UIImage *image = [BPKIcon templateIconNamed:iconName size:size];
+    
+    if ([BPKRTLSupport viewIsRTL:self] && flipsForRightToLeft) {
+        image = [image imageWithHorizontallyFlippedOrientation];
```

Comment:
Couldn't we just use `imageFlippedForRightToLeftLayoutDirection` instead? Then we could add another initializer to `BPKIcon` and this could would just be

```objective-c
    return [BPKIcon templateIconName:iconName size:size flipsForRightToLeftLayoutDirection:flipsForRightToLeft];
```
?

Reply:
`imageFlippedForRightToLeftLayoutDirection` soon will be deprecated: `This method cannot be used to create a left-to-right version of a right-to-left source image, and will be deprecated in a future release. New code should instead use -imageWithHorizontallyFlippedOrientation to construct a UIImageAsset.`

Reply:
Where did you see that? It's not in the docs

Reply:
Agree on that, however it's in the comments of UIImage.h right on the top of `- (UIImage *)imageFlippedForRightToLeftLayoutDirection NS_AVAILABLE_IOS(9_0);`

Reply:
Odd it's not in any of the docs, but it is in the code. Also images in assets catalogs support something similar. I am still inclined to use `imageFlippedForRightToLeftLayoutDirection` we can always change if it gets actually deprecated 🤔

Reply:
Switched back to `imageFlippedForRightToLeftLayoutDirection`, but didn't introduce a new factory method in BPKIcon as `templateIconName:size:` returns a `UIImage` which already has this feature.